rosin-project/rxros
# Migrate examples from rxros repository (and other suggestions)

As per subject.

This PR suggests a new repository layout, in which all examples are moved to the `rxros_examples` repository (rosin-project/rxros_examples#1), `rxros` is the only package in this repository and a new dependency is added: [rxcpp_vendor](https://github.com/rosin-project/rxcpp_vendor) (which packages in the RxCpp dependency).

Some other changes:

 - further streamlining of build scripts and package manifests
 - exploit (transitive) dependency declarations more (avoid restating primary dependencies in multiple places)
 - update readme with all these changes

To test this new setup, do the following:

```bash
source /opt/ros/$ROS_DISTRO/setup.bash
rosdep update
mkdir -p $HOME/rxros_refactor_ws
cd $HOME/rxros_refactor_ws
wstool init src https://raw.githubusercontent.com/gavanderhoorn/rxros_examples/refactor/rxros_examples.rosinstall
rosdep install --from-paths $HOME/rxros_refactor_ws/src --ignore-src
catkin_make  # or if you prefer: catkin build
source devel/setup.bash
```

This should build `rxcpp_vendor`, `rxros`, the BrickPi and the `teleop` packages/examples.

After `source`ing the workspace you should be able to run all of the examples.

Note: to initialise a workspace with just `rxros` (so none of the examples), use [this .rosinstall file](https://raw.githubusercontent.com/gavanderhoorn/rxros/refactor/rxros.rosinstall) instead of the one shown in the listing above.
